Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -331,6 +333,12 @@ public final class ManifestLoader: ManifestLoaderProtocol {
 
             // Check that the product references existing targets.
             for target in product.targets {
+                // Check if this target has the same name as an executable product name (by case-insensitive comparison)
+                if manifest.toolsVersion >= .v5_5,
+                   execNames.contains(target.lowercased()) {
```

Comment:
Are there issues like this for non-executable products as well?  If so, I can update this to check against all products.

Reply:
@neonichu I am also a little confused by the exact requirements here. isn't a target and product having the same name pretty typical? in many cases they are 1:1 and have the same name

Reply:
It's quite typical.  I don't think that this change is correct; I think that the conclusion from the bug report was that SwiftPM should guard against two targets have the same case-insensitive name, not whether a product and a target have the same name.

Reply:
Yep, that's right. We probably also should disallow having two products having the same name just differing by case, but a target and a product having the same name is OK (and even the default from `init` IIRC).

Reply:
Targets in the bug report don't have the same case-insensitive name (see the `testTargetNameAcrossProductDecls`). I'll look into an alternative way to resolve this, e.g. no --module-name for executables.